src/environment.go

```diff
@@ -56,6 +56,8 @@ type cache interface {
 	set(key, value string, ttl int)
 }
 
+type doGetRequestModifier func(request *http.Request)
```

Comment:
I like the idea, would rename to `HTTPRequestModifier`

Reply:
done - pushed.